src/compile/axis/encode.ts

```diff
@@ -27,8 +27,17 @@ export function labels(model: UnitModel, channel: SpatialScaleChannel, specified
     labelsSpec.text =  {
       signal: timeFormatExpression('datum.value', fieldDef.timeUnit, axis.format, config.axis.shortTimeLabels, config.timeFormat, isUTCScale)
     };
+  } else if ((fieldDef.type === NOMINAL || fieldDef.type === ORDINAL) && axis.format) {
+    if (fieldDef['formatType'] === 'number') {
+      labelsSpec.text = {
+        signal: `format(${fieldDef.field}, '${numberFormat(fieldDef, axis.format, config)}')`
+      };
+    } else if (fieldDef['formatType']) {
```

Comment:
This seems like timeFormat is the default, which is incorrect.  Number should be the default.